erpnext/hr/doctype/expense_claim/expense_claim.py

```diff
@@ -360,3 +361,43 @@ def get_expense_claim(
 	)
 
 	return expense_claim
+
+# api to create expense claim
+@frappe.whitelist()
+def create_expense_claim(subject, email, expense_date, expense_type, amount):
+	# fetch employee code from email
+	employee = frappe.get_all("Employee", or_filters={
+		"prefered_email": email,
+		"company_email": email,
+		"personal_email": email
+	})
+
+	if employee and employee[0]:
+		employee = frappe.get_doc("Employee", employee[0].name)
+	else:
+		frappe.throw(_("Employee not found."))
+
+	if employee.department:
+		department = frappe.get_doc("Department", employee.department)
+		if department.expense_approvers:
+			expense_approver = department.expense_approvers[ os.urandom(1)[0] % len(department.expense_approvers)].approver
+		else:
+			frappe.throw(_("Expense Approver not found for {0}").format(email))
+	else:
+		frappe.throw(_("Department is not set"))
+
+	# creating expense claim doc.
+	frappe.get_doc({
+		"doctype": "Expense Claim",
+		"employee": employee.name,
+		"payable_account": frappe.get_value('Company', employee.company, 'default_payable_account'),
+		"expense_approver": expense_approver,
+		"expenses": [
+				{
+					"expense_date": expense_date,
+					"expense_type": expense_type,
+					"amount": amount,
+					"description": subject
+				}
+			]
+	}).insert()
```

Comment:
do a ignore mandatory and ignore perms

Reply:
@neilLasrado If we ignore these, then someone still will have to go to the desk to complete the transaction. 

Are you thinking that the expense claim is created from the app even if there is partial information? And then, rest can be taken care from the desk? 

I was thinking that the API should raise error and fail gracefully so that we can show appropriate message to the user in the app.